attestation/jsonattestor/json.go

```diff
@@ -0,0 +1,213 @@
+package jsonattestor
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/cyberphone/json-canonicalization/go/src/webpki.org/jsoncanonicalizer"
+	"github.com/ghodss/yaml"
+	"github.com/itchyny/gojq"
+
+	"github.com/in-toto/go-witness/attestation"
+	"github.com/in-toto/go-witness/cryptoutil"
+	"github.com/in-toto/go-witness/log"
+	"github.com/in-toto/go-witness/registry"
+	"github.com/invopop/jsonschema"
+)
+
+const (
+	Name    = "json"
+	Type    = "https://witness.dev/attestations/json/v0.1"
+	RunType = attestation.PostProductRunType
+)
+
+// Attestor processes a JSON/YAML file, storing canonical JSON and user queries.
+type Attestor struct {
+	CanonicalJSON  []byte
```

Comment:
we should add json tags to the two exported variables to match our conventions in other attestors

`json:"canonicaljson"` and `json:"subjectqueries"`